_ml-commons-plugin/tutorials/bedrock-guardrails.md

```diff
@@ -0,0 +1,395 @@
+    }
+  ]
+}
+```
+
+### Step 4: Create a Claude model connector
+
+To use the guardrails with an Amazon Bedrock Claude model, first create a connector for the Claude endpoint:
+
+```json
+POST _plugins/_ml/connectors/_create
+{
+  "name": "BedRock claude Connector",
+  "description": "BedRock claude Connector",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "your_aws_region like us-east-1",
+    "service_name": "bedrock",
+    "anthropic_version": "bedrock-2023-05-31",
+    "max_tokens_to_sample": 8000,
+    "temperature": 0.0001,
+    "response_filter": "$.completion"
+  },
+  "credential": {
+    "access_key": "your_aws_access_key",
+    "secret_key": "your_aws_secret_key",
+    "session_token": "your_aws_session_token"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.us-east-1.amazonaws.com/model/anthropic.claude-v2/invoke",
+      "headers": {
+        "content-type": "application/json",
+        "x-amz-content-sha256": "required"
+      },
+      "request_body": "{\"prompt\":\"${parameters.prompt}\", \"max_tokens_to_sample\":${parameters.max_tokens_to_sample}, \"temperature\":${parameters.temperature},  \"anthropic_version\":\"${parameters.anthropic_version}\" }"
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 5: Register the Claude model
+
+Register the Claude model with input guardrails enabled. This configuration ensures that all requests to the model are first validated by the guardrail:
+
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+    "name": "Bedrock Claude V2 model",
+    "function_name": "remote",
+    "description": "Bedrock Claude V2 model",
+    "connector_id": "your_connector_id",
+    "guardrails": {
+        "input_guardrail": {
+            "model_id": "your_guardrail_model_id",
+            "response_filter":"$.action",
+            "response_validation_regex": "^\"NONE\"$"
+        },
+        "type": "model"
+    }
+}
+```
+{% include copy-curl.html %}
+
+### Step 6: Test the model
+
+First, test the model with acceptable inputs:
+
+```json
+POST /_plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "prompt": "\n\nHuman:${parameters.question}\n\nnAssistant:",
+    "question": "hello"
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response shows that the call was successful:
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "response",
+          "dataAsMap": {
+            "response": " Hello!"
+          }
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
+Next, test the model with inappropriate inputs:
+
+```json
+POST /_plugins/_ml/models/your_model_id/_predict
+{
+  "parameters": {
+    "prompt": "\n\nHuman:${parameters.question}\n\nnAssistant:",
+    "question": "how to rob a bank"
+  }
+}
+```
+{% include copy-curl.html %}
+
+Problematic inputs are blocked by the guardrail:
+
```

Comment:
Line 267: "guardrail" => either "guardrails" or "model". Or use "The response shows that the inappropriate input was blocked:" as on line 360.